### PR TITLE
Fixes the reloading issue

### LINKED ISSUE #23: Remember the selected theme
When developing, the test page reloads and we lose the selected theme.

### PR DESCRIPTION
**Pull request by iFlameing**
Fixes #23 

@domoritz can you review this?

### CODE REVIEW

**Comment by domoritz** [comment]
As an idea, instead of local storage, we should use the URL so that one can send a link to a theme demo from the deployed website you helped set up.

**Comment by iFlameing** [comment]
@domoritz should I have to store the same data in url which I am storing in localstorage?

**Comment by domoritz** [comment]
I imagine something like `...#dark` for the dark theme. Or `?theme=dark`. We wouldn't need localstorage anymore.

**Comment by iFlameing** [comment]
ok, I will try to implement it :)

**Comment by domoritz** [comment]
This PR needs some work before we can merge it. Ping me when it's ready.

**Comment by iFlameing** [comment]
@domoritz I implemented as you said. can you review this, please!

File: test/index.html [hunk 1]
```diff
@@ -75,8 +75,13 @@
  .catch(function(err) { console.error(err, err.stack); });
 
 var themes = document.querySelector('#themes');
```
Comment (domoritz) [inline test/index.html]: Please format with prettier.

File: test/index.html [hunk 2]
```diff
@@ -86,7 +91,23 @@
   refresh();
 });
 
+function getQueryVariable(variable)
```
Comment (domoritz) [inline test/index.html]: Is this really the cleanest way to do this? Is there no direct API to get variables from a URL?

File: test/index.html [hunk 3]
```diff
@@ -75,8 +75,13 @@
  .catch(function(err) { console.error(err, err.stack); });
 
 var themes = document.querySelector('#themes');
+var currentLocation = window.location;
+var url = new URL(currentLocation);
 themes.addEventListener('change', function() {
-  theme = vegaThemes[themes.options[themes.selectedIndex].value];
+  theme = vegaThemes[themes.options[themes.selectedIndex].value]; 
+  url.searchParams.set('id', `${themes.selectedIndex}`);
+  url.searchParams.set('themes', `${themes.options[themes.selectedIndex].value}`);
```
Comment (domoritz) [inline test/index.html]: Why do we need id and themes?

File: test/index.html (hunk 1 above)
Comment (iFlameing) [reply test/index.html]: ok

File: test/index.html (hunk 3 above)
Comment (iFlameing) [reply test/index.html]: we need Id for setting the value in selection tab and themes for updating the themes on the page.

Comment (domoritz) [reply test/index.html]: What is a selection tab?

Comment (iFlameing) [reply test/index.html]: Updating the value of a select tag. <select><select/> we can use only id but as you said that we have to include themes name in the url so I added the themes in url.

Comment (domoritz) [reply test/index.html]: When did I say we should include the theme name in the URL? I think a meaningful id is enough.

**Comment by iFlameing** [comment]
@domoritz can you take look, I pushed a new commit please!

**Comment by domoritz** [comment]
Note that I pushed some code yesterday to integrate your previous changes. Now this pr is quite large. What do we really need?

**Comment by iFlameing** [comment]
ok I will remove the unnecessary changes and make this pr smaller because the changes are not relevant to it. The pr become big because of prettier.

**Comment by iFlameing** [comment]
@domoritz I am closing this and opening a new one.

**Pull request closed without merging**